cpp/src/Slice/Grammar.y

```diff
@@ -1502,32 +1522,39 @@ interface_list
     StringTokPtr scoped = StringTokPtr::dynamicCast($1);
     ContainerPtr cont = unit->currentContainer();
     TypeList types = cont->lookupType(scoped->v);
-    if(!types.empty())
+    if (!types.empty())
     {
         InterfaceDeclPtr interface = InterfaceDeclPtr::dynamicCast(types.front());
-        if(!interface)
+        if (!interface)
         {
-            string msg = "`";
-            msg += scoped->v;
-            msg += "' is not an interface";
-            unit->error(msg);
+            if (auto alias = TypeAliasPtr::dynamicCast(types.front()))
+            {
+                if (!alias->typeMetadata().empty())
+                {
+                    unit->error("illegal metadata: typealias metadata `" + alias->typeMetadata().front() +
```

Comment:
Here, the check is in the wrong spot. The empty metadata list should be enforced when the alias is created.

Reply:
This check is specifically for inheritance, like while parsing:
`interface Foo : BarAlias`
In this specific case, obviously we don't allow metadata here:
`interface Foo : [metadata] Bar    // bogus`
And so the check makes sure the list is empty to enforce this.

But we don't know what context the typealias will be used in, and so the check has to be delayed until we have that context (after type-alias creation).
Or are you saying that we should disallow type-aliases of interfaces and classes to ever have metadata in any circumstance? (lumping in the above comment with this one)

Reply:
> Or are you saying that we should disallow type-aliases of interfaces and classes to ever have metadata in any circumstance? (lumping in the above comment with this one)

Yes. And more specifically:
a) the only metadata allowed in a typealias definition is type-metadata such as cs:type(), cpp:type() etc. Non-type metadata such as [amd] or [protected] or [cpp:const] (...) are not allowed.
b) for class and interface typealiases, we don't allow any metadata at all

Reply:
As of current metadata, B is already included in A, since there isn't any type metadata that can be applied to interfaces or classes that I'm aware of!

Reply:
I moved the check to when the type-alias is created, and added a check that forbids metadata to be specified on aliases of classes and interfaces.
It's on line 1685 of Grammar.y (I don't think I can link to lines in PRs on Github)